pkg/odo/cli/telemetry/telemetry.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+)
+
+const RecommendedCommandName = "telemetry"
+
+type TelemetryOptions struct {
+	telemetryData segment.TelemetryData
+}
+
+func NewTelemetryOptions() *TelemetryOptions {
+	return &TelemetryOptions{}
+}
+
+func (o *TelemetryOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	err = json.Unmarshal([]byte(args[0]), &o.telemetryData)
+	return err
+}
+
+func (o *TelemetryOptions) Validate() (err error) {
+	return err
+}
+
+func (o *TelemetryOptions) Run() (err error) {
+	var segmentClient *segment.Client
+	cfg, err := preference.New()
+	if err != nil {
+		return errors.Errorf("unable to update data, required preference.yaml file not found.")
+	}
+	// Initiate the segment client if ConsentTelemetry preference is set to true
+	if cfg.GetConsentTelemetry() {
+		if os.Getenv(segment.DisableTelemetryEnv) == "true" {
+			log.Warningf("Sending telemetry disabled by %s=%s\n", segment.DisableTelemetryEnv, os.Getenv(segment.DisableTelemetryEnv))
+		} else {
+			if segmentClient, err = segment.NewClient(cfg); err != nil {
+				klog.V(4).Infof("Cannot create a segment client, will not send any data: %s", err.Error())
+			}
+			defer segmentClient.Close()
+		}
+	}
+	if segmentClient != nil {
+		if serr := segmentClient.Upload(o.telemetryData); serr != nil {
+			klog.V(4).Infof("Cannot send data to telemetry: %q", serr)
+		}
+		return segmentClient.Close()
```

Comment:
We can put this in the `else` block above and avoid this check when GetConsentTelemetry() is false. We can also remove the `segmentClient` variable declaration above after this change.